src/de/jost_net/JVerein/gui/dialogs/BuchungenMitgliedskontenZuordnungDialog.java

```diff
@@ -0,0 +1,528 @@
+            if(assginMemberAccountToBooking(assignedBooking,  usedMemberAccount, dateFromInput, dateUntilInput, buchung, uniqueIbans.get(buchung.getIban()), "IBAN") ||
+                assginMemberAccountToBooking(assignedBooking, usedMemberAccount, dateFromInput, dateUntilInput, buchung, uniqueIds.get(bookingPurpose), externeMitgliedsnummer ? "Ext. Mitgliedsnummer" : "Mitgliedsnummer") ||
+                assginMemberAccountToBooking(assignedBooking, usedMemberAccount, dateFromInput, dateUntilInput, buchung, uniqueNames.get(bookingPurpose), "Vorname und Nachname")) 
+            {
+              continue;
+            }
+          }
+
+          if(assignedBooking.isEmpty())
+          {
+            GUI.getStatusBar().setErrorText("Es wurden keine passenden Buchungen oder Mitgliedskonten zum Zuordnen gefunden.");
+          }
+          else
+          {
+            BuchungenMitgliedskontenZuordnungVorschauDialog userValidationDialog = new BuchungenMitgliedskontenZuordnungVorschauDialog(assignedBooking);
+            userValidationDialog.open();
+          }
+        } 
+        catch (RemoteException e)
+        {
+          Logger.error("error while saving import file", e);
+          throw new ApplicationException("Fehler bei der Zuordnungssuche", e);
+        } 
+        catch (Exception e)
+        {
+          Logger.error("error while opening a dialog", e);
+          throw new ApplicationException("Fehler bei der Durchf�hrung der Zuordnung (Best�tigungsdialog)", e);
+        }
+      }
+
+      @Override
+      public void interrupt()
+      {
+        //
+      }
+
+      @Override
+      public boolean isInterrupted()
+      {
+        return false;
+      }
+    };
+
+    Application.getController().start(t);
+
+    close();
+  }
+
+  private String getBookingPurpose(Buchung buchung) throws RemoteException {
+    String zweck = buchung.getZweck();
+    if(zweck == null) return null;
+    zweck = zweck.replaceAll("\r\n", " ").replaceAll("\r", " ").replaceAll("\n", " ").toUpperCase();
+    // manche Banken haengen hier noch zusaetzliche Felder ran: diese versuchen wir Abzuschneiden
+    if(zweck.contains(" IBAN:")) {
+      zweck = zweck.substring(0, zweck.indexOf(" IBAN:"));
+    }
+    return zweck;
+  }
+
+  private boolean assginMemberAccountToBooking(final List<BookingMemberAccountEntry> assignedBooking, final Set<String> usedMemberAccount, Date dateFromInput, Date dateUntilInput, final Buchung buchung, String mitgliedsId, final String zuordnungsart)
+      throws RemoteException 
+  {
+    boolean processed = false;
+
+    if(mitgliedsId != null)
+    {
+      //wir wollen das nicht nochmal mit der Buchung probieren, da dies das gleiche Ergebnis w�re
+      processed = true;
+
+      ResultSetExtractor rs = new ResultSetExtractor()
+      {
+        @Override
+        public Object extract(ResultSet rs) throws SQLException, RemoteException
+        {
+          while (rs.next())
```

Comment:
Ich habe jetzt auch mal wieder etwas Zeit gefunden mir das anzusehen.

Es scheint hier so zu sein, das `rs.next()` immer false zurückliefert. Dadurch wird Zeile 359 nie ausgeführt und `assignedBooking` ist immer leer.

Reply:
Ja kann ich bestätigen. Hast du eine Idee wie man es umbauen kann?

Reply:
Leider noch nicht. Ich habe zwar schon etwas herumprobiert hatte aber keinen Erfolg. Mal sehen ob ich eine Lösung finde.

Reply:
Habe es abermals probiert. Es funktioniert schon. 
Es muss einen **Abrechnungslauf** für das Mitglied mit exaktem (Geld-)Betrag (wie die Buchung) und Stichtag, innerhalb des im Dialog gewählten Zeitraum, geben.

<img width="1412" alt="image" src="https://github.com/openjverein/jverein/assets/63780296/66921530-7863-44ba-b708-4ffff43d6801">

Reply:
Jetzt sieht es schon besser aus.

Was ich noch nicht hinbekommen habe ist eine Zuordnung mittels IBAN. Kannst du mal testen ob es funktioniert wenn nur ein Haken bei der IBAN ist.

Reply:
Die IBAN muss an der Buchung hängen. Sie wird nur bei der Übernahme aus Hibiscus gesetzt. Da Du vermutlich keine Buchung aus einem echten Umsatz hast, musst Du die IBAN in der Datenbank selbst eintragen. Dann sollte es gehen.

Reply:
Funktioniert mit beispielhaft hinterlegter IBAN im Mitglied und DB Eintrag in Buchung. Den Verwendungszweck habe ich dabei leer gelassen.
<img width="1403" alt="image" src="https://github.com/openjverein/jverein/assets/63780296/de59a457-ac2a-4407-aac2-697315fec185">